### PR TITLE
G. Shapiro - JS III Project - All Problems Other Than Stretch

### PR DESCRIPTION
**Pull request by lex-node**
solves all problems in JS III Project other than stretch problem

### CODE REVIEW

File: assignments/prototypes.js [hunk 1]
```diff
@@ -16,13 +16,33 @@
...
 
+function GameObject(attributes) {
+    this.createdAt = attributes.createdAt;
+    this.name = attributes.name;
+    this.dimensions = attributes.dimensions;
+    this.destroy = function () {
+        return `${this.name} was removed from the game`;
+    }
+}
+
```
Comment (jacobAngulo) [inline assignments/prototypes.js:27]: I could be mistaken but I was under the impression that you were supposed to add methods to your object constructors using this method, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/prototype#Examples

I'm not sure what the pro's and cons of this are but I'm sure it's worth looking up

File: assignments/prototypes.js [hunk 2]
```diff
@@ -41,70 +72,137 @@
...
+Villain.protoype = Object.create(Humanoid.prototype);
+
+function Hero(attributes)  {
+    Humanoid.call(this, attributes);
+    this.ninjaChop = function(obj){
+        obj.healthPoints -= 5;
+        console.log(`${this.name} has attacked ${obj.name} with a heroic ninja chop! costs 5 healthpoints!`);
+        if(obj.healthPoints <= 0) {
```
Comment (jacobAngulo) [inline assignments/prototypes.js:162]: I can't tell why but, batman is not ninja-chopping the joker according to my developer console

**Review by jacobAngulo — APPROVED** [review]
Nice work man! tell Gal that I'm available if yall break up

File: assignments/prototypes.js (hunk 1 above)
Comment (lex-node) [reply assignments/prototypes.js:27]: Yep, I think you're right. I realized that during the sprint yesterday and figured out the prototype method on the fly. Think I've got it down now...= )

**Pull request merged**